api/v1alpha1/contour_types.go

```diff
@@ -49,13 +49,13 @@ type ContourSpec struct {
 	//
 	// +kubebuilder:default=2
 	// +kubebuilder:validation:Minimum=0
-	Replicas int32 `json:"replicas"`
+	Replicas int32 `json:"replicas,omitempty"`
 
 	// namespace defines the schema of a Contour namespace.
 	// See each field for additional details.
 	//
-	// +optional
-	Namespace *NamespaceSpec `json:"namespace,omitempty"`
+	// +kubebuilder:default={name: "projectcontour", removeOnDeletion: false}
```

Comment:
do we need to specify the defaults both here and on the fields within `NamespaceSpec`?

Reply:
@skriss it's not required, but I think it's helpful when reading the code or the CRD. I can remove in a follow-on PR if you prefer.

Reply:
not a big deal, i think it can stay as-is.